src/lib/Microsoft.Health.Events/Telemetry/Exceptions/EventHubExceptionTelemetryProcessor.cs

```diff
@@ -0,0 +1,70 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Net.Sockets;
+using Azure.Messaging.EventHubs;
+using EnsureThat;
+using Microsoft.Health.Common.Telemetry;
+using Microsoft.Health.Logging.Telemetry;
+
+namespace Microsoft.Health.Events.Telemetry.Exceptions
+{
+    public enum EventHubErrorCode
+    {
+        /// <summary>
+        /// Error code that categorizes exceptions of the type EventHubsException
+        /// </summary>
+        OperationErrors,
+
+        /// <summary>
+        /// Error code that indicates failures in initializing event hub partition
+        /// </summary>
+        EventHubPartitionInitFailed,
+
+        /// <summary>
+        /// Error code that categorizes exceptions of the type SocketException
+        /// </summary>
+        SocketErrors,
+
+        /// <summary>
+        /// Error code that categorizes all other generic Exceptions
+        /// </summary>
+        GeneralErrors,
+    }
+
+    public static class EventHubExceptionTelemetryProcessor
```

Comment:
nit: Style should be in separate files.  Surprised this isn't being flag by code style rules.

Reply:
Moved it, although by default enums are allowed - https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1402.md